tools/import_diagnostics.py

```diff
@@ -110,17 +198,33 @@ def zigpy_device_from_legacy_diagnostics(
     for epid, ep in cluster_data.items():
         endpoint.request = AsyncMock(return_value=[0])
         for cluster_id, cluster in ep["in_clusters"].items():
-            real_cluster = device.endpoints[int(epid)].in_clusters[int(cluster_id, 16)]
+            try:
+                real_cluster = device.endpoints[int(epid)].in_clusters[
+                    int(cluster_id, 16)
+                ]
+            except KeyError:
+                _LOGGER.warning(
+                    "Failed to find server cluster 0x%s on endpoint %s: %s",
+                    cluster_id,
+                    epid,
+                    cluster,
+                )
+                continue
+
             if patch_cluster:
                 patch_cluster_for_testing(real_cluster)
             for attr_id, attr in cluster["attributes"].items():
                 if (
-                    attr["value"] is None
+                    attr.get("value", None) is None
```

Comment:
nit: `None` should already be the default:
```suggestion
                    attr.get("value") is None
```